packages/replay/src/upload.ts

```diff
@@ -151,14 +162,54 @@ class ReplayClient {
     }
   }
 
-  async connectionEndRecordingUpload(recordingId: string) {
-    if (!this.client) throw new Error("Protocol client is not initialized");
+  async uploadPart(link: string, part: any, size: number): Promise<string> {
+    return new Promise((resolve, reject) => {
+      console.log("Stating worker");
+      const worker = new Worker(path.join(__dirname, "./uploadWorker.js"));
 
-    await this.client.sendCommand<{ recordingId: string }>("Internal.endRecordingUpload", {
-      recordingId,
+      worker.on("message", resolve);
+      worker.on("error", reject);
+      worker.on("exit", code => {
+        if (code !== 0) {
+          reject(new Error(`Worker stopped with exit code ${code}`));
+        }
+      });
+
+      worker.postMessage({ link, part, size });
     });
   }
 
+  async uploadRecordingInParts(path: string, partUploadLinks: string[], partSize: number) {
+    // TODO: Can we not use this? It can use high memory
+    const fileBuffer = fs.readFileSync(path);
```

Comment:
Yeah, I think this will be problematic to keep all in memory. We should use streams instead I think